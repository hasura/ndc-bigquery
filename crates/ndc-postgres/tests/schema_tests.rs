pub mod common;

#[tokio::test]
async fn get_schema() {
    let result = common::get_schema().await;
    insta::assert_json_snapshot!(result);
}
