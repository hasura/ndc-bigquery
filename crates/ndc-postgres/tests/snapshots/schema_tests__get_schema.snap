---
source: crates/ndc-postgres/tests/schema_tests.rs
expression: result
---
{
  "scalar_types": {
    "any": {
      "aggregate_functions": {},
      "comparison_operators": {},
      "update_operators": {}
    }
  },
  "object_types": {
    "Album": {
      "fields": {
        "AlbumId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "ArtistId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Title": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "Artist": {
      "fields": {
        "ArtistId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Name": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "Customer": {
      "fields": {
        "Address": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "City": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Company": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Country": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "CustomerId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Email": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Fax": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "FirstName": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "LastName": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Phone": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "PostalCode": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "State": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "SupportRepId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "Employee": {
      "fields": {
        "Address": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "BirthDate": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "City": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Country": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Email": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "EmployeeId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Fax": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "FirstName": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "HireDate": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "LastName": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Phone": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "PostalCode": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "ReportsTo": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "State": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Title": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "Genre": {
      "fields": {
        "GenreId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Name": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "Invoice": {
      "fields": {
        "BillingAddress": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "BillingCity": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "BillingCountry": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "BillingPostalCode": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "BillingState": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "CustomerId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "InvoiceDate": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "InvoiceId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Total": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "InvoiceLine": {
      "fields": {
        "InvoiceId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "InvoiceLineId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Quantity": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "TrackId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "UnitPrice": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "MediaType": {
      "fields": {
        "MediaTypeId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Name": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "Playlist": {
      "fields": {
        "Name": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "PlaylistId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "PlaylistTrack": {
      "fields": {
        "PlaylistId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "TrackId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "Track": {
      "fields": {
        "AlbumId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Bytes": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Composer": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "GenreId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "MediaTypeId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Milliseconds": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "Name": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "TrackId": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "UnitPrice": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "geography_columns": {
      "fields": {
        "coord_dimension": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "f_geography_column": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "f_table_catalog": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "f_table_name": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "f_table_schema": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "srid": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "type": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "geometry_columns": {
      "fields": {
        "coord_dimension": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "f_geometry_column": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "f_table_catalog": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "f_table_name": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "f_table_schema": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "srid": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "type": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    },
    "spatial_ref_sys": {
      "fields": {
        "auth_name": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "auth_srid": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "proj4text": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "srid": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        },
        "srtext": {
          "arguments": {},
          "type": {
            "type": "named",
            "name": "any"
          }
        }
      }
    }
  },
  "collections": [
    {
      "name": "Album",
      "arguments": {},
      "type": "Album",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Album": {
          "unique_columns": [
            "AlbumId"
          ]
        }
      },
      "foreign_keys": {
        "FK_AlbumArtistId": {
          "column_mapping": {
            "ArtistId": "ArtistId"
          },
          "foreign_collection": "Album"
        }
      }
    },
    {
      "name": "Artist",
      "arguments": {},
      "type": "Artist",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Artist": {
          "unique_columns": [
            "ArtistId"
          ]
        }
      },
      "foreign_keys": {}
    },
    {
      "name": "Customer",
      "arguments": {},
      "type": "Customer",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Customer": {
          "unique_columns": [
            "CustomerId"
          ]
        }
      },
      "foreign_keys": {
        "FK_CustomerSupportRepId": {
          "column_mapping": {
            "SupportRepId": "EmployeeId"
          },
          "foreign_collection": "Customer"
        }
      }
    },
    {
      "name": "Employee",
      "arguments": {},
      "type": "Employee",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Employee": {
          "unique_columns": [
            "EmployeeId"
          ]
        }
      },
      "foreign_keys": {
        "FK_EmployeeReportsTo": {
          "column_mapping": {
            "ReportsTo": "EmployeeId"
          },
          "foreign_collection": "Employee"
        }
      }
    },
    {
      "name": "Genre",
      "arguments": {},
      "type": "Genre",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Genre": {
          "unique_columns": [
            "GenreId"
          ]
        }
      },
      "foreign_keys": {}
    },
    {
      "name": "Invoice",
      "arguments": {},
      "type": "Invoice",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Invoice": {
          "unique_columns": [
            "InvoiceId"
          ]
        }
      },
      "foreign_keys": {
        "FK_InvoiceCustomerId": {
          "column_mapping": {
            "CustomerId": "CustomerId"
          },
          "foreign_collection": "Invoice"
        }
      }
    },
    {
      "name": "InvoiceLine",
      "arguments": {},
      "type": "InvoiceLine",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_InvoiceLine": {
          "unique_columns": [
            "InvoiceLineId"
          ]
        }
      },
      "foreign_keys": {
        "FK_InvoiceLineInvoiceId": {
          "column_mapping": {
            "InvoiceId": "InvoiceId"
          },
          "foreign_collection": "InvoiceLine"
        },
        "FK_InvoiceLineTrackId": {
          "column_mapping": {
            "TrackId": "TrackId"
          },
          "foreign_collection": "InvoiceLine"
        }
      }
    },
    {
      "name": "MediaType",
      "arguments": {},
      "type": "MediaType",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_MediaType": {
          "unique_columns": [
            "MediaTypeId"
          ]
        }
      },
      "foreign_keys": {}
    },
    {
      "name": "Playlist",
      "arguments": {},
      "type": "Playlist",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Playlist": {
          "unique_columns": [
            "PlaylistId"
          ]
        }
      },
      "foreign_keys": {}
    },
    {
      "name": "PlaylistTrack",
      "arguments": {},
      "type": "PlaylistTrack",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_PlaylistTrack": {
          "unique_columns": [
            "PlaylistId",
            "TrackId"
          ]
        }
      },
      "foreign_keys": {
        "FK_PlaylistTrackPlaylistId": {
          "column_mapping": {
            "PlaylistId": "PlaylistId"
          },
          "foreign_collection": "PlaylistTrack"
        },
        "FK_PlaylistTrackTrackId": {
          "column_mapping": {
            "TrackId": "TrackId"
          },
          "foreign_collection": "PlaylistTrack"
        }
      }
    },
    {
      "name": "Track",
      "arguments": {},
      "type": "Track",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Track": {
          "unique_columns": [
            "TrackId"
          ]
        }
      },
      "foreign_keys": {
        "FK_TrackAlbumId": {
          "column_mapping": {
            "AlbumId": "AlbumId"
          },
          "foreign_collection": "Track"
        },
        "FK_TrackGenreId": {
          "column_mapping": {
            "GenreId": "GenreId"
          },
          "foreign_collection": "Track"
        },
        "FK_TrackMediaTypeId": {
          "column_mapping": {
            "MediaTypeId": "MediaTypeId"
          },
          "foreign_collection": "Track"
        }
      }
    },
    {
      "name": "geography_columns",
      "arguments": {},
      "type": "geography_columns",
      "deletable": false,
      "uniqueness_constraints": {},
      "foreign_keys": {}
    },
    {
      "name": "geometry_columns",
      "arguments": {},
      "type": "geometry_columns",
      "deletable": false,
      "uniqueness_constraints": {},
      "foreign_keys": {}
    },
    {
      "name": "spatial_ref_sys",
      "arguments": {},
      "type": "spatial_ref_sys",
      "deletable": false,
      "uniqueness_constraints": {
        "spatial_ref_sys_pkey": {
          "unique_columns": [
            "srid"
          ]
        }
      },
      "foreign_keys": {}
    }
  ],
  "functions": [],
  "procedures": []
}
